src/Sentry/Internal/DebugStackTrace.cs

```diff
@@ -0,0 +1,508 @@
+        if (match.Success && match.Groups.Count == 2)
+        {
+            frame.Function = match.Groups[1].Value + " { <lambda> }";
+        }
+    }
+
+    /// <summary>
+    /// Remove return type from module in a Task with a Lambda with a return value.
+    /// This was seen in Unity, see https://github.com/getsentry/sentry-unity/issues/845
+    /// </summary>
+    private static void DemangleLambdaReturnType(SentryStackFrame frame)
+    {
+        if (frame.Module == null)
+        {
+            return;
+        }
+
+        // Change:
+        //   System.Threading.Tasks.Task`1[[System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]] in InnerInvoke
+        //   or System.Collections.Generic.List`1[[UnityEngine.Events.PersistentCall, UnityEngine.CoreModule, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null]] in get_Item
+        // to:
+        //   System.Threading.Tasks.Task`1 in InnerInvoke`
+        //   or System.Collections.Generic.List`1 in get_Item
+        var match = RegexAsyncReturn.Match(frame.Module);
+        if (match.Success && match.Groups.Count == 2)
+        {
+            frame.Module = match.Groups[1].Value;
+        }
+    }
+
+#if ANDROID
+    private IAndroidAssemblyReader? _assemblyReader;
+    private bool _assemblyReaderInitialized = false;
+    private object _assemblyReaderLock = new();
+
+    private void InitializeAssemblyReader()
+    {
+        if (_assemblyReaderInitialized)
+        {
+            return;
+        }
+        lock (_assemblyReaderLock)
+        {
+            // check again after acquiring the lock
+            if (_assemblyReaderInitialized)
+            {
+                return;
+            }
+
+            var apkPath = Environment.CommandLine;
+            try
+            {
+                if (!File.Exists(apkPath))
+                {
+                    _options.DiagnosticLogger?.LogWarning("Cannot create AssemblyReader: cannot read APK path from Environment.CommandLine={0}", apkPath);
+                }
+                else
+                {
+#pragma warning disable CS0618 // Type or member is obsolete
+                    var supportedAbis = AndroidBuild.SupportedAbis ?? new List<string> { AndroidBuild.CpuAbi ?? "" };
+#pragma warning restore CS0618 // Type or member is obsolete
+                    _assemblyReader = AndroidAssemblyReaderFactory.Open(apkPath, supportedAbis, _options.DiagnosticLogger);
+                }
+            }
+            catch (Exception e)
+            {
+                _options.DiagnosticLogger?.LogError("Cannot create AssemblyReader: {0}", e);
+            }
+            finally
+            {
+                _assemblyReaderInitialized = true;
+            }
+        }
+    }
+#endif
+
+    private PEReader? TryReadAssembly(string assemblyName)
+    {
+#if ANDROID
+        InitializeAssemblyReader();
+        return _assemblyReader?.TryReadAssembly(assemblyName);
+#else
```

Comment:
This doesn't work in the Unity SDK when building for Android - we always get the non-android variant. I assume it's because we pre-compile a single DLL before including in the game. Looks like we would need runtime checks instead? That also plays into the approach to take about including AndroidAssemblyReader and the LZ4 dependency.

cc @bitsandfoxes since you haven't been active in this PR yet

Reply:
The Android support done in .NET doesn't really apply to Unity. They have their own Android support stuff, so we only use `netstandard2.0` target (and DLL) from this code base.

Reply:
In this case it does apply (or rather it would if it worked), because this would be executed for Mono builds where it's would help get line numbers. See https://github.com/getsentry/sentry-unity/pull/1106

Reply:
Anything we have in sentry-unity for Android doesn't use the Android support in sentry-dotnet.  Sentry-Unity only uses the `netstandard2.0` target from sentry-dotnet.  If we want to consolidate platform support, we'll have to look at that separately.